doc/howto.md

```diff
@@ -0,0 +1,428 @@
+(aside from its didactic use). In this section we are going to create a simple Southbound interface. It's important
+to remark that the nature of the Southbound API itself has nothing to do with the creation process of an IoT Agent.
+Each device protocol will use its own mechanisms and it is up to the IoTA developer to find any libraries that would 
+help him in its development. In this example, we will use Express as such library. 
+
+In order to add the Express dependency to your project, add the following line to the `dependencies` section of the
+`package.json`:
+```
+    "express": "*",
+```
+And install the dependencies as usual with `npm install`. You will have to require both `express` and `http` in
+your code as well.
+
+Now, in order to accept connections in our code, we have to start express first. With this purpose in mind, we will
+create a new functino `initSouthbound()`, that will be called from the initialization code of our IOTA:
+```
+function initSouthbound(callback) {
+    southboundServer = {
+        server: null,
+        app: express(),
+        router: express.Router()
+    };
+
+    southboundServer.app.set('port', 8080);
+    southboundServer.app.set('host', '0.0.0.0');
+
+    southboundServer.router.get('/iot/d', manageULRequest);
+    southboundServer.server = http.createServer(southboundServer.app);
+    southboundServer.app.use('/', southboundServer.router);
+    southboundServer.server.listen(southboundServer.app.get('port'), southboundServer.app.get('host'), callback);
+}
+```
+
+This Express code set up a HTTP server, listening in the 8080 port, that will handle incoming requests targeting 
+path `/iot/d` using the middleware `manageULRequest()`. This middleware will contain all the Southbound logic, and 
+the library methods we need in order to progress the information to the Context Broker. The code of this middleware
+would be as follows:
+
+```
+function manageULRequest(req, res, next) {
+    var values;
+
+    iotAgentLib.getDevice(req.query.i, function(error, device) {
+        if (error) {
+            res.status(404).send({
+                message: 'Couldn\'t find the device: ' + JSON.stringify(error)
+            });
+        } else {
+            values = parseUl(req.query.d, device);
+            iotAgentLib.update(device.name, device.type, '', values, device, function(error) {
+                if (error) {
+                    res.status(500).send({
+                        message: 'Error updating the device'
+                   });
+                } else {
+                    res.status(200).send({
+                        message: 'Device successfully updated'
+                    });
+                }        
+            });
+        }
+    });  
+}
+```
+
+For this middleware we have made use of a function `parseUl()` that parses the data payload and transforms it 
+in the data map expected by the update function:
+```
```

Comment:
I still prefer you to just mention that there are other occurrences of some issue :D PRs end up being really difficult to read if they have two or three comments repeated once and again.